src/Service.Host/client/src/components/goodsIn/GoodsInUtility.js

```diff
@@ -123,15 +126,18 @@ function GoodsInUtility({
 
     useEffect(() => {
         if (bookInResult?.message) {
-            setMessage({ error: false, text: bookInResult.message, success: bookInResult.success });
+            setMessage({ error: !bookInResult?.success, text: bookInResult.message });
```

Comment:
What does the '!' do here? Also do we need the '?' before success since you've checked bookInResult at the if above?

Reply:
it's an error if not success, hence the !. And yeah I think that ? is redundant

Reply:
Ahhh yeah of course, for some reason I wasn't reading it as a "not". D'oh!